src/CMakeLists.txt

```diff
@@ -0,0 +1,172 @@
+if ( ${T8CODE_ENABLE_DYNAMIC_LINKAGE} )
+    add_library( T8 SHARED )
+    set_target_properties( T8 PROPERTIES POSITION_INDEPENDENT_CODE ON )
+else()
+    add_library( T8 STATIC )
+endif()
+
+set_target_properties( T8 PROPERTIES OUTPUT_NAME t8 )
+
+target_compile_definitions( T8 PUBLIC T8_CMAKE_BUILD )
+target_compile_definitions( T8 PUBLIC T8_CC="${CMAKE_C_COMPILER}" )
+target_compile_definitions( T8 PUBLIC T8_CFLAGS="${CMAKE_C_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_CPP="${CMAKE_CXX_COMPILER}" )
+target_compile_definitions( T8 PUBLIC T8_CPPFLAGS="${CMAKE_CXX_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_LDFLAGS="${CMAKE_SHARED_LINKER_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_LIBS="???" )
+
+# There has to be an elegant solution to embed the version and commit here...
+target_compile_definitions( T8 PUBLIC T8_PACKAGE_STRING="t8 1.6.1.???" )
+target_compile_definitions( T8 PUBLIC T8_VERSION="1.6.1.???" )
+target_compile_definitions( T8 PUBLIC T8_VERSION_MAJOR=1 )
+target_compile_definitions( T8 PUBLIC T8_VERSION_MINOR=6 )
+target_compile_definitions( T8 PUBLIC T8_VERSION_POINT=1.??? )
+
+target_include_directories( T8 PUBLIC ${CMAKE_CURRENT_LIST_DIR} )
+target_link_libraries( T8 PUBLIC P4EST::P4EST SC::SC )
+
+if ( T8CODE_ENABLE_MPI )
+    target_compile_definitions( T8 PUBLIC T8_ENABLE_MPI )
+    target_compile_definitions( T8 PUBLIC T8_ENABLE_MPIIO )
+    target_link_libraries( T8 PUBLIC MPI::MPI_C )
+endif()
+
```

Comment:
Is it very tedious to also add options to link t8code against VTK?

A typical `configure` call to link against VTK looks like this:
```
  ../t8code/configure \
    --prefix="${T8CODE_INSTALL_PREFIX}" \
    --enable-mpi \
    --with-vtk \
    --with-vtk_version_number=9.1 \
    CFLAGS="-O3" CXXFLAGS="-O3" \
    CC=mpicc CXX=mpicxx \
    LDFLAGS="-L${VTK_INSTALL_PREFIX}/lib" \
    CPPFLAGS="-I${VTK_INSTALL_PREFIX}/include/vtk-9.1"
```

Reply:
I do not have VTK installed on my machine, but according to VTK's doc it should work now. Can you test on your side?